### PR TITLE
Add first version of the Storyscript service

### PR DESCRIPTION
**Pull request by wilzbach**
From https://github.com/storyscript/storyscript/issues/417.

- [ ] `omg exec version` currently fails for unknown reasons (still needs to be debugged, running the docker image manually works fine -  maybe it's https://github.com/microservices/omg-cli/issues/99)
- [ ] `omg exec grammar` depends on the grammar being exposed via the API (though, of course, we could call `storyscript grammar` as a workaround for now)
- [ ]  `omg exec compile ...` depends on https://github.com/storyscript/storyscript/pull/414

Other TODO:
- [ ] don't use the Flask development mode, but a production server (what are our guidelines/preference here)
- [ ] add Travis CI testing for a basic `omg exec`

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,9 @@
+FROM        python:3.7-alpine
+
+RUN         mkdir /app
+ADD         requirements.txt /app
+RUN         pip install -r /app/requirements.txt
+RUN         pip install https://github.com/storyscript/storyscript/archive/master.zip
```
Comment (judepereira) [inline Dockerfile]: Let's version control this. storyscript==0.7.1 for now.

File: README.md [hunk 2]
```diff
@@ -1,2 +1,10 @@
 # storyscript
 An OMG server exposing the Storyscript compiler
+
+Usage
+-----
+
+```coffee
+# Storyscript
+storyscript compile files: {'my.story': 'x = 2'}
```
Comment (judepereira) [inline README.md]: Can you add other examples here? How to compile one line, or parse, lex etc.

File: app.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        # TODO: use https://github.com/storyscript/storyscript/pull/414 here
+        filename, value = files.popitem()
+        return jsonify(storyscript.loads(value))
+
+    @app.route('/compile', methods=['POST'])
+    def compile():
+        files = request.json
+        # TODO: use https://github.com/storyscript/storyscript/pull/414 here
+        filename, value = files.popitem()
+        return jsonify(storyscript.loads(value))
```
Comment (judepereira) [inline app.py]: Can you handle errors? Send across a response with the exception message (if any), and wrap the response in something like the following:
{success: true/false, output: {<actual Storyscript compile output goes here>}}

File: Dockerfile (hunk 1 above)
Comment (wilzbach) [reply Dockerfile]: Why 0.7.1 and not 0.8.1? (the latest one)

File: README.md (hunk 2 above)
Comment (wilzbach) [reply README.md]: Added lex & parse, but not grammar and version as they are boring and just provided for completeness.

File: Dockerfile (hunk 1 above)
Comment (judepereira) [reply Dockerfile]: Engine is currently on 0.7.1. When I upgrade the engine, we'll update this to be 0.8.1.

Also, once the Hub starts supporting versions for services, then this version requirement is moot as every Storyscript version can be in the Hub, and I can control the Storyscript version in the deploy story.

Comment (wilzbach) [reply Dockerfile]: Downgraded to 0.7.1

File: app.py (hunk 3 above)
Comment (wilzbach) [reply app.py]: There's a global error handling that catches all errors and sends them as JSON to the user.

**Comment by judepereira** [comment]
Waiting on https://github.com/storyscript/storyscript/issues/449

**Comment by wilzbach** [comment]
Looks like there are still issues with OMG CLI:

```
Successfully tagged omg/wilzbach/omg-storyscript:latest
Exited with code 1
```

It works locally with Postman.
Example input:

```json
{"files": {"foo.story": "x=2"}}
```

### /lex

```json
{
    "foo.story": [
        [
            0,
            "NAME",
            "x"
        ],
        [
            1,
            "EQUALS",
            "="
        ],
        [
            2,
            "INT",
            "2"
        ]
    ]
}
```

### /parse

```json
{
    "foo.story": {
        "block": {
            "rules": {
                "assignment": {
                    "assignment_fragment": {
                        "expression": {
                            "multiplication": {
                                "exponential": {
                                    "factor": {
                                        "entity": {
                                            "values": {
                                                "number": {
                                                    "value": "2"
                                                }
                                            }
                                        }
                                    }
                                }
                            }
                        },
                        "value": "="
                    },
                    "path": {
                        "value": "x"
                    }
                }
            }
        }
    }
}
```

### compile

```json
{
    "entrypoint": [
        "foo.story"
    ],
    "services": [],
    "stories": {
        "foo.story": {
            "entrypoint": "1",
            "functions": {},
            "modules": {},
            "services": [],
            "tree": {
                "1": {
                    "args": [
                        2
                    ],
                    "command": null,
                    "enter": null,
                    "exit": null,
                    "function": null,
                    "ln": "1",
                    "method": "set",
                    "name": [
                        "x"
                    ],
                    "output": null,
                    "parent": null,
                    "service": null
                }
            },
            "version": "0.9.0"
        }
    }
}
```

**Pull request merged**